### PR TITLE
Dependent Field Import Optimization, Code Prepend Support

### PR DESCRIPTION
**Pull request by Hrishabh17**
## Summary by CodeRabbit

- **New Features**
  - Introduced a parameter to prepend codes to names across multiple modules.
  - Added new functions to disable categories, cost centers, and projects, enhancing management capabilities.
  - Implemented robust error handling and logging for import processes.
  - Enhanced synchronization processes for various entities, ensuring data integrity.
  - Improved flexibility in attribute handling during synchronization.
  - New functions for batch processing of cost codes and types, improving efficiency.
  - Added functionality to check project synchronization eligibility based on recent activity.

- **Bug Fixes**
  - Improved exception handling and logging in synchronization methods.

- **Chores**
  - Updated dependency versions to incorporate new features and fixes.

### CODE REVIEW

File: apps/mappings/imports/queues.py [hunk 1]
```diff
@@ -47,6 +49,8 @@ def chain_import_fields_to_fyle(workspace_id):
             workspace_id,
             destination_field,
             'CATEGORY',
+            False,
+            True if 'ACCOUNT' in import_code_fields else False,
```
Comment (ashwin1111) [inline apps/mappings/imports/queues.py:53]: added comment to slack

File: apps/sage_intacct/models.py [hunk 2]
```diff
@@ -1501,7 +1505,7 @@ def bulk_create_or_update(cost_types: List[Dict], workspace_id: int):
                 cost_types_to_be_created.append(cost_type_object)
 
             elif cost_type['RECORDNO'] in primary_key_map.keys() and (
-                cost_type['NAME'] != primary_key_map[cost_type['RECORDNO']]['name'] or cost_type['STATUS'] != primary_key_map[cost_type['RECORDNO']]['status']
+                cost_type['STATUS'] != primary_key_map[cost_type['RECORDNO']]['status']
```
Comment (ashwin1111) [inline apps/sage_intacct/models.py:1508]: we don't disable things as well, why are updates required?

File: apps/sage_intacct/utils.py [hunk 3]
```diff
@@ -23,17 +23,25 @@
...
     'projects': 25000,
     'customers': 15000,
-    'items':15000,
+    'items': 15000,
     'classes': 15000
 }
 
+ATTRIBUTE_DISABLE_CALLBACK_PATH = {
+    'PROJECT': 'apps.mappings.imports.modules.projects.disable_projects',
+    'ACCOUNT': 'apps.mappings.imports.modules.categories.disable_categories'
```
Comment (ashwin1111) [inline apps/sage_intacct/utils.py:41]: expense type/gl account

File: apps/mappings/imports/modules/cost_centers.py [hunk 4]
```diff
@@ -111,7 +111,6 @@ def disable_cost_centers(workspace_id: int, cost_centers_to_disable: Dict, *args
         if code:
             payload = {
                 'name': expense_attribute.value,
-                'code': code,
```
Comment (ashwin1111) [inline apps/mappings/imports/modules/cost_centers.py]: let's not update any existing features, someone might complaint that this worked in past :)

**Review by ashwin1111 — APPROVED** [review]

File: apps/mappings/imports/modules/cost_centers.py (hunk 4 above)
Comment (Hrishabh17) [reply apps/mappings/imports/modules/cost_centers.py]: yes, removed code from cost center payload.

**Pull request merged**